ui/src/flux/containers/FluxConnectionPage.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import {withRouter} from 'react-router'
+
+import FluxNew from 'src/flux/components/FluxNew'
+import FluxEdit from 'src/flux/components/FluxEdit'
+import PageHeader from 'src/reusable_ui/components/page_layout/PageHeader'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import {getService} from 'src/shared/apis'
+import {FluxFormMode} from 'src/flux/constants/connection'
+
+import {
+  updateServiceAsync,
+  UpdateServiceAsync,
+  createServiceAsync,
+  CreateServiceAsync,
+  fetchFluxServicesForSourceAsync,
+  FetchFluxServicesForSourceAsync,
+  setActiveServiceAsync,
+  SetActiveServiceAsync,
+} from 'src/shared/actions/services'
+import {couldNotGetFluxService} from 'src/shared/copy/notifications'
+import {notify as notifyAction} from 'src/shared/actions/notifications'
+import {Service, Source, Notification} from 'src/types'
+
+interface Props {
+  source: Source
+  services: Service[]
+  params: {id: string; sourceID: string}
+  router: {push: (url: string) => void}
+  notify: (message: Notification) => void
+  createService: CreateServiceAsync
```

Comment:
this was an approach i took when first typing DashboardPage, but i think @chnn has an improved and simpler pattern that you could consider here, if i understand it correctly, which is that you don't need a new type definition and instead can just use `typeof createServiceAsync` for all of these

Reply:
nice thanks!